kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -206,6 +219,188 @@ func checkResourcesStatus(src *v1alpha1.KafkaSource) error {
+		if scalingClass == kedaAutoscalingAnnotationClass {
+			r.cooldownPeriod = convertMapKeyToInt32(annotations, kedaCooldownPeriodAnnodationKey, logger)
+			r.pollingInterval = convertMapKeyToInt32(annotations, kedaPollingIntervalAnnodationKey, logger)
+			r.pollingInterval = convertMapKeyToInt32(annotations, kedaPollingIntervalAnnodationKey, logger)
+			r.triggerLagThreshold = convertMapKeyToInt32(annotations, kedaTriggerLagThresholdAnnodationKey, logger)
+
+		}
+	}
+}
+
+func (r *Reconciler) generateKedaScaledObjectUnstructured(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) (*unstructured.Unstructured, error) {
+	logger := logging.FromContext(ctx).Desugar()
+	deploymentName := ra.GetName()
+	namespace := src.Namespace
+	name := src.Name
+	logger.Info("Unstructured ScaledObject name "+name, zap.Any("name", name))
+	srcName := src.GetName()
+	srcUID := src.GetUID()
+	srcResVersion := src.GetResourceVersion()
+	logger.Info("Got srcResVersion="+srcResVersion, zap.Any("srcResVersion", srcResVersion))
+	srcKind := src.GetGroupVersionKind().Kind
+	logger.Info("Got srcKind srcName srcUID", zap.Any("srcKind", srcKind), zap.Any("srcName", srcName), zap.Any("srcUID", srcUID))
+	srcBrokerList := src.Spec.BootstrapServers
+	srcConcumerGroup := src.Spec.ConsumerGroup
+	triggers := make([]map[string]interface{}, 0, 1)
+	topics := strings.Split(src.Spec.Topics, ",")
+	if len(topics) == 0 {
+		return nil, fmt.Errorf("Comma-separated list of topics can not be empty")
+	}
+	for _, topic := range topics {
+		triggerMetadata := map[string]interface{}{
+			"brokerList":    srcBrokerList,
+			"consumerGroup": srcConcumerGroup,
+			"topic":         topic,
+		}
+		if r.triggerLagThreshold != nil {
+			logger.Info("Got triggerLagThreshold", zap.Any("triggerLagThreshold", r.triggerLagThreshold))
+			triggerMetadata["lagThreshold"] = strconv.Itoa(int(*r.triggerLagThreshold))
+		}
+		trigger := map[string]interface{}{
+			"type":     "kafka",
+			"metadata": triggerMetadata,
+		}
+		triggers = append(triggers, trigger)
+	}
+	spec := map[string]interface{}{
+		"scaleTargetRef": map[string]interface{}{
+			"deploymentName": deploymentName,
+		},
+		"triggers": triggers,
+	}
+	if r.minReplicaCount != nil {
+		logger.Info("Got minReplicaCount", zap.Any("minReplicaCount", r.minReplicaCount))
+		spec["minReplicaCount"] = *r.minReplicaCount
+	}
+	if r.maxReplicaCount != nil {
+		logger.Info("Got maxReplicaCount", zap.Any("maxReplicaCount", r.maxReplicaCount))
+		spec["maxReplicaCount"] = *r.maxReplicaCount
+	}
+	if r.cooldownPeriod != nil {
+		logger.Info("Got cooldownPeriod", zap.Any("cooldownPeriod", r.cooldownPeriod))
+		spec["cooldownPeriod"] = *r.cooldownPeriod
+	}
+	if r.pollingInterval != nil {
+		logger.Info("Got pollingInterval", zap.Any("pollingInterval", r.minReplicaCount))
+		spec["pollingInterval"] = *r.pollingInterval
+	}
+	soUnstr := &unstructured.Unstructured{
```

Comment:
I think there should be a resources folder, where you have a scaled_object.go file to create this... That is typically the convention we follow, but not sure if you guys are doing so in kafka.. See example: https://github.com/google/knative-gcp/blob/26efd4cf70d23a7d0fde2231d941608ebf8da98b/pkg/reconciler/pubsub/pullsubscription/keda/resources/scaled_object.go#L40

Reply:
Yes for future -wanted to start small with minimal changes

Reply:
this would match with the labels, receive_adapter, and eventtypes patter, I'd recommend doing so with the `scaled_object.go` file as well